package zurg;

option cc_generic_services = true;
option java_generic_services = true;
option py_generic_services = true;
option java_package = "com.chenshuo.muduo.zurg.proto";
option java_outer_classname = "MasterProto";

import "muduo/protorpc2/rpc2.proto";

message SlaveHeartbeat {
  required string slave_name = 1;
  required int64 send_time_us = 2;

  optional string host_name = 20;
  optional int32 listen_port = 21;
  optional int32 slave_pid = 22;
  optional int64 start_time_us = 23;
}

service MasterService {
  rpc slaveHeartbeat (SlaveHeartbeat) returns (rpc2.Empty) {
    option (rpc2.noreturn) = true;
  }
}

